tgg/seqr_scripts/sex_check.py

```diff
@@ -0,0 +1,199 @@
+        chr_place = 22
+    else:
+        try:
+            chr_place = int(chr_name) - 1
+        except:
+            logger.warning("chr_name cannot be converted to an integer")
+
+    #could utilize hail's reference genome chr names
+    chr_name= hl.get_reference(build).contigs[chr_place]
+
+    logging.info(f"Filtering to non-par regions on {chr_name} to calculate normalized {chr_name} coverage...")
+    sex_mt = hl.filter_intervals(mt, [hl.parse_locus_interval(chr_name, reference_genome=build)])
+    if chr_place == 22 or chr_place == 23 :
+        sex_mt = sex_mt.filter_rows((filter_nonpar_expr), keep = True)
+
+
+    #filter to common snvs above defined callrate (because only biallelic, should now only have one index in the array)
+    sex_mt = sex_mt.filter_rows(sex_mt.AF > 0.01)
+
+    #optional for gnomad methods
+    sex_mt = hl.variant_qc(sex_mt) 
+    sex_mt = sex_mt.filter_rows(sex_mt.variant_qc.call_rate > call_rate_threshold)
+
+    logging.info(f"Calculating mean coverage on {chr_name}...")
+    sex_mt = sex_mt.annotate_cols(**{f"{chr_name}_mean_dp": mt.aggregate(hl.agg.mean(mt.DP))})
+    sex_ht = sex_mt.cols()
+
+    return(sex_ht)
+
+def run_hails_impute_sex(mt: hl.MatrixTable, 
+    build: str ,
+    outdir: str, 
+    callset_name: str,
+    male_fstat_threshold: float=0.75, 
+    female_fstat_threshold: float=0.5, 
+    aaf_threshold: float=0.05) -> hl.Table:
+    """
+    Imputes sex and annotates MatrixTable with results, outputs a histogram of fstat values
+    :param mt: MatrixTable containing samples to be ascertained for sex
+    :param build: Reference used, either GRCh37 or GRCh38
+    :param callset_name: Basename for callset and output results
+    :param outdir: Directory to output results
+    :param male_fstat_threshold: Fstat threshold above which a sample will be called male
+    :param female_fstat_threshold: Fstat threshold below which a sample will be called female
+    :param aaf_threshold: Minimum alternate allele frequency required 
+    :return: Table with imputed sex annotations
+    """
+    
+    logging.warn("User needs to decide on fstat thresholds for male/female by looking at fstat plots")
+
+    
+    # filter to the X chromosome and impute sex
+    mt = hl.filter_intervals(mt, [hl.parse_locus_interval(get_reference_genome(mt.locus).x_contigs[0], reference_genome=build)])   
+    sex_ht = hl.impute_sex(mt.GT, aaf_threshold=aaf_threshold, male_threshold=male_fstat_threshold, female_threshold=female_fstat_threshold)
+    mt = mt.annotate_cols(**sex_ht[mt.col_key])
+    sex_ht = mt.cols()
+    
+    # plot histogram of fstat values
+    df = sex_ht.to_pandas()
+    plt.clf()
+    plt.hist(df["f_stat"])
+    plt.xlabel("Fstat")
+    plt.ylabel("Frequency")
+    plt.axvline(male_fstat_threshold, color='blue', linestyle='dashed', linewidth=1)
+    plt.axvline(female_fstat_threshold, color='red', linestyle='dashed', linewidth=1)
+
+    outplot = f"{outdir}/fstat_{callset_name}.png"
```

Comment:
it would be helpful to have the user be able to input a desired output path rather than creating it here (this is optional for moving the code into this repo, but we should add it if we move the code into gnomad_methods)